storage/tests/unit/test_blob.py

```diff
@@ -1472,7 +1475,8 @@ def _do_resumable_upload_call2(
             'PUT', resumable_url, data=payload, headers=expected_headers)
 
     def _do_resumable_helper(
-            self, use_size=False, num_retries=None, predefined_acl=None):
+            self, use_size=False, num_retries=None, predefined_acl=None,
+            extra_headers=None):
```

Comment:
We should add a testcase, e.g. `test__do_resumable_upload_with_headers`, passing actual headers.  E.g.:

```python
    def test__do_resumable_upload_with_retry(self):
        self._do_resumable_helper(extra_headers={'X-Goog-Testing': 'true'})
```